R/sim_fixed_n.R

```diff
@@ -405,12 +406,12 @@ doAnalysis <- function(d, rho_gamma, n_stratum) {
 
   event <- sum(d$event)
   if (n_stratum > 1) {
-    ln_hr <- survival::coxph(survival::Surv(tte, event) ~ (treatment == "experimental") + survival::strata(stratum), data = d)$coefficients
+    ln_hr <- survival::coxph(Surv(tte, event) ~ (treatment == "experimental") + strata(stratum), data = d)$coefficients
```

Comment:
My only lingering concern is that this line that uses `survival::strata()` is not covered by our tests/examples, and this usage is specifically noted as behaving incorrectly in previous versions of {survival}. From the release notes:

> Note that prior to this update using survival::strata(inst) above would give the wrong answer; because :: mucks up formula processing the term would not be recognized as a stratum, and the resulting fit equivalent to using factor(inst).

My questions are:

1. Can we add a test/example with more than one stratum to execute this code?
2. Are we worried that this could give incorrect results when run with survival <3.8-1?

Reply:
Current coverage: https://app.codecov.io/github/Merck/simtrial/blob/main/R%2Fsim_fixed_n.R#L408

![image](https://github.com/user-attachments/assets/c46c105f-fbde-4cee-8e08-5815eb7106c4)